File: Wallet.ts
// Wallet.ts

export class Wallet {
    private balance: number;
    private transactions: string[];

    constructor() {
        this.balance = 0;
        this.transactions = [];
    }

    addIncome(amount: number): void {
        this.balance += amount;
        this.transactions.push(`Income: +$${amount}`);
    }

    addExpense(amount: number): void {
        if (amount > this.balance) {
            console.log("Insufficient funds!");
        } else {
            this.balance -= amount;
            this.transactions.push(`Expense: -$${amount}`);
        }
    }

    checkBalance(): string {
        return `Current Balance: $${this.balance}`;
    }

    viewTransactions(): string {
        if (this.transactions.length === 0) {
            return "No transactions yet.";
        }
        return this.transactions.join("\n");
    }
}

File: wallet.py
# wallet.py

class Wallet:
    def __init__(self):
        self.balance = 0
        self.transactions = []

    def add_income(self, amount):
        self.balance += amount
        self.transactions.append(f"Income: +${amount}")

    def add_expense(self, amount):
        if amount > self.balance:
            print("Insufficient funds!")
        else:
            self.balance -= amount
            self.transactions.append(f"Expense: -${amount}")

    def check_balance(self):
        return f"Current Balance: ${self.balance}"

    def view_transactions(self):
        if not self.transactions:
            return "No transactions yet."
        return "\n".join(self.transactions)
